File: F#/Shared/UserModels.fs
module UserModels

    open Newtonsoft.Json
    open System.Runtime.Serialization
    
    [<DataContract;CLIMutable>]
    [<JsonObject(MemberSerialization=MemberSerialization.OptOut)>]
    type User = 
        { Email : string 
          FirstName : string
          LastName : string }
        static member CreateUser(?Email, ?FirstName, ?LastName) = 
            let initMember x ``default`` = x |> Option.fold (fun state param->param) ``default``
            { Email = initMember Email ""
              FirstName = initMember FirstName ""
              LastName = initMember LastName "" }
        
    [<DataContract;CLIMutable>]
    [<JsonObject(MemberSerialization=MemberSerialization.OptOut)>]
    type AccountResponse =
        {
            [<DataMember(Name="success")>]
            Success : bool
            [<DataMember(Name="error")>]
            Error : string
            [<DataMember(Name="user")>]
            User : User
            [<DataMember(Name="token")>]
            Token : string
        }
        static member CreateAccountResponse(?Success, ?Error, ?UserParam, ?Token) = 
            let initMember x ``default`` = x |> Option.fold (fun state param->param) ``default``
            { Success = initMember Success false
              Error = initMember Error ""
              User = initMember UserParam (User.CreateUser())
              Token = initMember Token ""}

File: Shared/SSO/UserModels.cs
using System;
using System.Text;
using Newtonsoft.Json;

namespace Xamarin.SSO.Client
{
    [JsonObject (MemberSerialization.OptIn)]
    public class User
    {
        [JsonProperty (PropertyName = "email")]
        public string Email { get; set; }

        [JsonProperty (PropertyName = "firstname")]
        public string FirstName { get; set; }

        [JsonProperty (PropertyName = "lastname")]
        public string LastName { get; set; }
    }

    [JsonObject (MemberSerialization.OptIn)]
    public class AccountResponse
    {
        [JsonProperty (PropertyName = "success")]
        public bool Success { get; set; }
        [JsonProperty (PropertyName = "error")]
        public string Error { get; set; }
        [JsonProperty (PropertyName = "user")]
        public User User { get; set; }
        [JsonProperty (PropertyName = "token")]
        public string Token { get; set; }
        // Expiration info?
    }

    [Flags]
    public enum Access
    {
        None = 0,
        Admin = 1,
        Write = 1 << 1,
        Read = 1 << 2,
    }
}
